src/Microsoft.Azure.ServiceBus/Message.cs

```diff
@@ -328,7 +394,13 @@ public DateTime LockedUntilUtc
                 internal set => this.lockedUntilUtc = value;
             }
 
-            /// <summary>Gets the unique number assigned to a message by Service Bus, for this entity.</summary>
+            /// <summary>Gets the unique number assigned to a message by Service Bus.</summary>
+            /// <remarks>
+            ///     The sequence number is a unique 64-bit integer assigned to a message as it is accepted 
+            ///     and stored by the broker and functions as its true identifier. For partitioned entities, 
+            ///     the topmost 16 bits reflect the partition identifier. Sequence numbers monotonically increase 
+            ///     and are gapless. They roll over to 0 when the 48-64 bit range is exhausted. This property is read-only.
```

Comment:
>and are gapless. [](start = 20, length = 16)

There can be perceived gaps in the case of say ScheduledMessages. Though there won't be gaps on the server side, the fact might not hold true for the receivers. Should we remove this?

Reply:
Scheduled messages get sequenced as they get put into the queue. How is there a gap?

Reply:
@clemensv, Lets say we schedule a message. It gets enqueued as sequence number 10. And then when the message is actually pushed to the active queue, we get the next active sequence number and stamp it (say 15). 
For a customer who doesn't know or care about the internals of broker, he would see a missing sequence number of 10. 
Similar things happens with Deduplication feature as well. The duplicated message gets its own sequence number, but is never inserted into the queue. So the customer will never see that sequence number.
Adding "gapless" to the contract enforces us to follow it and taking that contract out is a difficult challenge. I think we should just stick to "monotonically increasing numbers" and not commit ourselves to "gapless"